src/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.js

```diff
@@ -83,12 +85,15 @@ export default class rd2ElevateCreditCardForm extends LightningElement {
     set paymentMethod(value) {
         this.notifyIframePaymentMethodChanged(value);
         this._paymentMethod = value;
+        this.enableWidgetOnPaymentMethodChange = true;
     }
 
     notifyIframePaymentMethodChanged(newValue) {
         const iframe = this.selectIframe();
         if(this.shouldNotifyIframe(newValue)) {
```

Comment:
I think having this `shouldNotifyIframe` boolean logic inside of the notify function could lead to confusion since `notifyIframePaymentMethodChanged()` indicates that the notification is going to happen without condition. It might be better to wrap `notifyIframePaymentMethodChanged()` inside of `if (this.shouldNotifyIframe()` in the `paymentMethod` setter so that it reads a little clearer. Think we can run into issues down the road when we have conditionals inside functions when both are at different levels of abstraction since in terms of the call flow, checking if we need to send the notification comes before sending it and is not actually a part of sending it.

Reply:
That makes sense, that whole function is wrapped in a conditional minus a single line.